packages/htmlbars-compiler/lib/hydration-opcode-compiler.js

```diff
@@ -121,19 +114,17 @@ HydrationOpcodeCompiler.prototype.closeElement = function() {
 
 HydrationOpcodeCompiler.prototype.mustache = function(mustache) {
   this.pushMorphPlaceholderNode();
-  
-  var sexpr = mustache.sexpr;
 
   var morphNum = this.morphNum++;
   var start = this.currentDOMChildIndex;
   var end = this.currentDOMChildIndex;
   this.morphs.push([morphNum, this.paths.slice(), start, end, mustache.escaped]);
 
-  if (isHelper(sexpr)) {
-    prepareSexpr(this, sexpr);
+  if (isHelper(mustache)) {
+    prepareCommonOpcodes(this, mustache);
```

Comment:
I'd prefer to just expand this out to the 3 calls so that it stays at the same abstraction as the AST.